packages/backend/src/workers/provider/LlamaCppPython.ts

```diff
@@ -63,33 +76,75 @@ export class LlamaCppPython extends InferenceProvider {
 
     const envs: string[] = [`MODEL_PATH=/models/${modelInfo.file.file}`, 'HOST=0.0.0.0', 'PORT=8000'];
     envs.push(...getModelPropertiesForEnvironment(modelInfo));
-    const isMacGPU = vmType === VMType.LIBKRUN;
-    if (isMacGPU) {
-      envs.push('GPU_LAYERS=99');
-    }
 
-    const devices = isMacGPU
-      ? [
-          {
+    const mounts: MountConfig = [
+      {
+        Target: '/models',
+        Source: modelInfo.file.path,
+        Type: 'bind',
+      },
+    ];
+
+    const deviceRequests: DeviceRequest[] = [];
+    const devices: Device[] = [];
+    let entrypoint: string | undefined = undefined;
+    let cmd: string[] = [];
+    let user: string | undefined = undefined;
+
+    if (gpu) {
+      // mounting
+
+      switch (vmType) {
+        case VMType.WSL:
+          mounts.push({
+            Target: '/usr/lib/wsl',
+            Source: '/usr/lib/wsl',
+            Type: 'bind',
+          });
+
+          devices.push({
+            PathOnHost: '/dev/dxg',
+            PathInContainer: '/dev/dxg',
+            CgroupPermissions: 'r',
+          });
+
+          user = '0';
+
+          cmd = [
+            '-c',
+            '/usr/bin/ln -s /usr/lib/wsl/lib/* /usr/lib64/ && PATH="${PATH}:/usr/lib/wsl/lib/" && chmod 755 ./run.sh && ./run.sh',
+          ];
+          break;
+        case VMType.LIBKRUN:
+          devices.push({
             PathOnHost: '/dev/dri',
             PathInContainer: '/dev/dri',
-          },
-        ]
-      : [];
+            CgroupPermissions: 'r',
+          });
+          break;
+      }
+
+      // adding gpu capabilities
+      deviceRequests.push({
+        Capabilities: [['gpu']],
+        Count: -1, // -1: all
+      });
+
+      entrypoint = '/usr/bin/sh';
```

Comment:
This breaks the execution when using the vulkan image. The service is not start at all, no logs. Once i remove the entrypoint it starts working fine again

Reply:
Oh yeah thanks  :pray:  for noticing, could you retry now @lstocchi ?